flow/layers/layer_raster_cache_item.cc

```diff
@@ -0,0 +1,176 @@
+    }
+  }
+}
+
+std::optional<RasterCacheKeyID> LayerRasterCacheItem::GetId() const {
+  switch (cache_state_) {
+    case kCurrent:
+      return key_id_;
+    case kChildren: {
+      return layer_children_id_;
+    }
+    default:
+      return {};
+  }
+}
+
+bool LayerRasterCacheItem::Rasterize(const PaintContext& paint_context,
+                                     SkCanvas* canvas) const {
+  FML_DCHECK(cache_state_ != CacheState::kNone);
+  SkISize canvas_size = canvas->getBaseLayerSize();
+  SkNWayCanvas internal_nodes_canvas(canvas_size.width(), canvas_size.height());
+  internal_nodes_canvas.setMatrix(canvas->getTotalMatrix());
+  internal_nodes_canvas.addCanvas(canvas);
+  PaintContext context = {
+      // clang-format off
+          .internal_nodes_canvas         = static_cast<SkCanvas*>(&internal_nodes_canvas),
+          .leaf_nodes_canvas             = canvas,
+          .gr_context                    = paint_context.gr_context,
+          .dst_color_space               = paint_context.dst_color_space,
+          .view_embedder                 = paint_context.view_embedder,
+          .raster_time                   = paint_context.raster_time,
+          .ui_time                       = paint_context.ui_time,
+          .texture_registry              = paint_context.texture_registry,
+          .raster_cache                  = paint_context.raster_cache,
+          .checkerboard_offscreen_layers = paint_context.checkerboard_offscreen_layers,
+          .frame_device_pixel_ratio      = paint_context.frame_device_pixel_ratio,
+      // clang-format on
+  };
+
+  switch (cache_state_) {
+    case CacheState::kCurrent:
+      FML_DCHECK(layer_->needs_painting(context));
+      layer_->Paint(context);
+      break;
+    case CacheState::kChildren:
+      FML_DCHECK(layer_->as_container_layer());
+      layer_->as_container_layer()->PaintChildren(context);
+      break;
+    case CacheState::kNone:
+      FML_DCHECK(cache_state_ != CacheState::kNone);
+      return false;
+  }
+  return true;
+}
+
+static const auto* flow_type = "RasterCacheFlow::Layer";
+
+bool LayerRasterCacheItem::TryToPrepareRasterCache(const PaintContext& context,
+                                                   bool parent_cached) const {
+  if (!context.raster_cache || parent_cached) {
+    return false;
+  }
+  if (cache_state_ != kNone) {
+    RasterCache::Context r_context = {
+        // clang-format off
+      .gr_context         = context.gr_context,
+      .dst_color_space    = context.dst_color_space,
+      .matrix             = matrix_,
+      .logical_rect       = layer_->paint_bounds(),
+      .flow_type          = flow_type,
+      .checkerboard       = context.checkerboard_offscreen_layers,
+        // clang-format on
+    };
+    return context.raster_cache->UpdateCacheEntry(
+        GetId().value(), r_context,
+        [=](SkCanvas* canvas) { Rasterize(context, canvas); });
```

Comment:
Please explicitly capture what you need. In this case, I would consider making `Rasterize` a static method and passing in `cache_state_` to it.

Reply:
Ditto

Reply:
About this I had change the lambda blocked value to `[ctx = context, this]`

Reply:
Please avoid capturing this

Reply:
fixed